server/src/test/java/org/elasticsearch/indices/cluster/IndicesClusterStateServiceRandomUpdatesTests.java

```diff
@@ -390,12 +397,19 @@ public ClusterState randomlyUpdateClusterState(ClusterState state,
             IndicesClusterStateService indicesClusterStateService = clusterStateServiceMap.get(node);
             MockIndicesService indicesService = (MockIndicesService) indicesClusterStateService.indicesService;
             for (MockIndexService indexService : indicesService) {
-                for (MockIndexShard indexShard : indexService) {
-                    ShardRouting persistedShardRouting = indexShard.routingEntry();
-                    if (persistedShardRouting.initializing() && randomBoolean()) {
-                        startedShards.add(persistedShardRouting);
-                    } else if (rarely()) {
-                        failedShards.add(new FailedShard(persistedShardRouting, "fake shard failure", new Exception(), randomBoolean()));
+                final Index index = indexService.index();
+                // do not start or fail shards of indices that were just closed or reopened, because
+                // they are still initializing and we must wait for the cluster state to be applied
+                // on node before starting or failing them
```

Comment:
I'm confused. The cluster state was already applied on the node, no? I don't understand the extra restriction here.